markdown_it/main.py

```diff
@@ -27,36 +27,51 @@
     linkify_it = None
 
 
-_PRESETS = AttrDict(
-    {
-        "default": presets.default.make(),
-        "zero": presets.zero.make(),
-        "commonmark": presets.commonmark.make(),
-    }
-)
+_PRESETS = {
+    "default": presets.default.make(),
+    "zero": presets.zero.make(),
+    "commonmark": presets.commonmark.make(),
+}
+_PRESETS["js-default"] = _PRESETS["default"]
+_PRESETS["gfm-like"] = presets.default.make()
+_PRESETS["gfm-like"]["options"]["linkify"] = True
 
 
 class MarkdownIt:
     def __init__(
-        self, config: Union[str, Mapping] = "commonmark", renderer_cls=RendererHTML
+        self,
+        config: Union[str, Mapping] = "commonmark",
+        options_update: Optional[Mapping] = None,
```

Comment:
One neat trick I've recently started to use in my APIs is
```python
from types import MappingProxyType

# This is basically just an immutable empty mapping
EMPTY_MAP = MappingProxyType({})

def func(options: Mapping = EMPTY_MAP):
    ...
```

This convention saves me a bunch of `Optional` annotations and `None` handling, and avoids the dangerous default problem.

Could be applied here if you like it :smile: You basically just need to create one global `EMPTY_MAP` instance per project and can reuse it.

Reply:
thanks for the tip, I will have a look but probably not add it in this PR